typings/index.d.ts

```diff
@@ -0,0 +1,2022 @@
+}
+
+declare module 'material-ui/Dialog/Dialog' {
+  import { TransitionHandlers } from 'material-ui/internal/Transition';
+
+  export interface DialogProps extends Partial<TransitionHandlers> {
+    fullScreen?: boolean;
+    ignoreBackdropClick?: boolean;
+    ignoreEscapeKeyUp?: boolean;
+    enterTransitionDuration?: number;
+    leaveTransitionDuration?: number;
+    maxWidth?: 'xs' | 'sm' | 'md';
+    onBackdropClick?: Function;
+    onEscapeKeyUp?: Function;
+    onRequestClose?: React.EventHandler<any>;
+    open?: boolean;
+    transition?: Function | React.ReactElement<any>;
+  }
+
+  export default class Dialog extends MaterialUI.Component<DialogProps> {}
+}
+
+declare module 'material-ui/Dialog/DialogActions' {
+  export interface DialogActionsProps {}
+
+  export default class DialogActions extends MaterialUI.Component<
+    DialogActionsProps
+  > {}
+}
+
+declare module 'material-ui/Dialog/DialogContent' {
+  export interface DialogContentProps {}
+
+  export default class DialogContent extends MaterialUI.Component<
+    DialogContentProps
+  > {}
+}
+
+declare module 'material-ui/Dialog/DialogContentText' {
+  export interface DialogContentTextProps {}
+
+  export default class DialogContentText extends MaterialUI.Component<
+    DialogContentTextProps
+  > {}
+}
+
+declare module 'material-ui/Dialog/DialogTitle' {
+  export interface DialogTitleProps {
+    disableTypography?: boolean;
+  }
+
+  export default class DialogTitle extends MaterialUI.Component<
+    DialogTitleProps
+  > {}
+}
+
+declare module 'material-ui/Dialog/withResponsiveFullScreen' {
+  import { Breakpoint } from 'material-ui/styles/breakpoints';
+  import { WithWidthEnhancement } from 'material-ui/utils/withWidth';
+
+  export interface WithResponsiveFullScreenOptions {
+    breakpoint: Breakpoint;
+  }
+
+  export default function withResponsiveFullScreen<P>(
+    options: WithResponsiveFullScreenOptions
+  ): React.ComponentClass<P & WithWidthEnhancement>;
+}
+
+declare module 'material-ui/Divider' {
+  export { default } from 'material-ui/Divider/Divider';
+  export * from 'material-ui/Divider/Divider';
+}
+
+declare module 'material-ui/Divider/Divider' {
+  export interface DividerProps {
+    absolute?: boolean;
+    inset?: boolean;
+    light?: boolean;
+  }
+
+  export default class Divider extends MaterialUI.Component<DividerProps> {}
+}
+
+declare module 'material-ui/Drawer' {
+  export { default } from 'material-ui/Drawer/Drawer';
+  export * from 'material-ui/Drawer/Drawer';
+}
+
+declare module 'material-ui/Drawer/Drawer' {
+  import { Theme } from 'material-ui/styles/theme';
+
+  export interface DrawerProps {
+    anchor?: 'left' | 'top' | 'right' | 'bottom';
+    docked?: boolean;
+    elevation?: number;
+    enterTransitionDuration?: number;
+    leaveTransitionDuration?: number;
+    onRequestClose?: React.EventHandler<any>;
+    open?: boolean;
+    SlideProps?: Object;
+    theme: Theme;
```

Comment:
Thanks for the typings! This doesn't look right though, theme param and its not optional?